safehttp/response_writer.go

```diff
@@ -64,21 +66,68 @@ func NotWritten() Result {
 	return Result{}
 }
 
-// Write TODO
+// Write dispatches the response to the Dispatcher, setting the Content-Type to
+// text/html; charset=utf-8 and response status to 200 OK if resp is a safe
+// response. The Dispatcher will then write the response to the underlying
+// Response Writer.
+//
+// TODO: replace panics with proper error handling when getting the response
+// Content-Type or writing the response fails.
 func (w *ResponseWriter) Write(resp Response) Result {
-	w.markWritten()
+	ct, err := w.d.ContentType(resp)
+	if err != nil {
+		panic(err)
+	}
+	w.rw.WriteHeader(int(StatusOK))
+	w.rw.Header().Set("Content-Type", ct)
 	if err := w.d.Write(w.rw, resp); err != nil {
-		panic("error")
+		panic(err)
 	}
+	w.markWritten()
```

Comment:
Why this change?

Reply:
We decided in the simplifying session to only call `markWritten` after a response has been written and this change made its way in this PR as I was refactoring things.

Reply:
Please leave out this change for a separate PR.

Reply:
The big problem with this is that it introduces a bug, making the tests  panic. So, if we put `markWritten` at the beginning that will set `w.written` to `true`. Then if either `d.ContentType(...)` or `d.WriteSomething(...)` returns an error, we will panic and that will be recovered by the `ServeMux`. The `ServeMux` will then attempt to call`WriteError` with a 500, which attempts to call `markWritten` again, which will panic because `w.written` was already set to true which ends up in a recursive thing, I think :/

Reply:
Therefore, I'd suggest leaving it as it is in this PR and then fixing it properly in a follow-up PR (the alternative is `t.Skip()` the tests in `integration/mux/mux_test.go` that test the behaviour of passing unsafe responses until we fix it properly)

Reply:
Opened #164.